impeller/renderer/backend/vulkan/command_pool_vk.cc

```diff
@@ -52,6 +48,9 @@ std::shared_ptr<CommandPoolVK> CommandPoolVK::GetThreadLocal(
 }
 
 void CommandPoolVK::ClearAllPools(const ContextVK* context) {
+  if (tls_command_pool.get()) {
+    tls_command_pool.get()->erase(context->GetHash());
```

Comment:
Move this to the end of the function so it runs after the block that calls `Reset` on every `CommandPoolVK` associated with this context.

In theory this should be unnecessary because nobody else should be holding on to a strong reference to this thread's `CommandPoolVK`, and the `erase` should destruct it and do all the necessary cleanup.  But for safety I think if would be prudent to ensure that the current thread's `CommandPoolVK` for this context is explicitly reset at the same time as the ones for other threads.

Reply:
Actually I just realized this is unnecessary.  If this thread's `CommandPoolVK` is still alive after the `tls_command_pool.erase`, then the corresponding `weak_ptr` in `g_all_pools` will still be promotable to a `shared_ptr` and will get the `Reset` call.

So the current implementation in the PR will work as intended.